vulcanai/datasets/utils.py

```diff
@@ -66,83 +67,42 @@ def check_split_ratio(split_ratio):
-#     for example in examples:
-#         strata_maps[getattr(example, strata_field)].append(example)
-#     return list(strata_maps.values())
-
-
-# def rationed_split(examples, train_ratio, test_ratio, val_ratio, rnd):
-#     """
-#     Create a random permutation of examples, then split them
-#     by ratio x length slices for each of the train/test/dev? splits
-#     :param examples:
-#     :param train_ratio: Train ratio
-#     :param test_ratio: Test ratio
-#     :param val_ratio: Val ratio
-#     :param rnd:
-#     :return:
-#     """
-#     # Create a random permutation of examples, then split them
-#     # by ratio x length slices for each of the train/test/dev? splits
-#     N = len(examples)
-#     randperm = rnd(range(N))
-#     train_len = int(round(train_ratio * N))
-#
-#     # Due to possible rounding problems
-#     if not val_ratio:
-#         test_len = N - train_len
-#     else:
-#         test_len = int(round(test_ratio * N))
-#
-#     indices = (randperm[:train_len],  # Train
-#                randperm[train_len:train_len + test_len],  # Test
-#                randperm[train_len + test_len:])  # Validation
-#
-#     # There's a possibly empty list for the validation set
-#     data = tuple([examples[i] for i in index] for index in indices)
-#
-#     return data
-
-# class RandomShuffler(object):
-#     """Use random functions while keeping track of the random state to
-# make it
-#     reproducible and deterministic."""
-#
-#     def __init__(self, random_state=None):
-#         self._random_state = random_state
-#         if self._random_state is None:
-#             self._random_state = random.getstate()
-#
-#     @contextmanager
-#     def use_internal_state(self):
-#         """Use a specific RNG state."""
-#         old_state = random.getstate()
-#         random.setstate(self._random_state)
-#         yield
-#         self._random_state = random.getstate()
-#         random.setstate(old_state)
-#
-#     @property
-#     def random_state(self):
-#         return deepcopy(self._random_state)
-#
-#     @random_state.setter
-#     def random_state(self, s):
-#         self._random_state = s
-#
-#     def __call__(self, data):
-#         """Shuffle and return a new list."""
-#         with self.use_internal_state():
-#             return random.sample(data, len(data))
+
+def rationed_split(df, train_ratio, test_ratio, validation_ratio):
+    """
+    Function to split a dataset given ratios
+
+    Parameters:
+        df: Dataframe
+            The dataframe you want to split
+        train_ratio: int
+            proportion of the dataset that will go to the train split.
+            between 0 and 1
+        test_ratio: int
+            proportion of the dataset that will go to the test split.
+            between 0 and 1
+        validation_ratio: int
+            proportion of the dataset that will go to the val split.
+            between 0 and 1
+
+    Returns:
+        indices: tuple of list of indices.
+    """
```

Comment:
a quick test or assertion should be done that the ratios add to 1

Reply:
Done before in check_split_ratio when called from split in tabular dataset, but the contract is not explicit. I'd rather modularize and leave that up to the calling function but can be persuaded otherwise. For now, have added docs saying that the check is not performed.

Reply:
Done before in check_split_ratio when called from split in tabular dataset, but the contract is not explicit. I'd rather modularize and leave that up to the calling function but can be persuaded otherwise. For now, have added docs saying that the check is not performed.

Reply:
Oops I must’ve missed it. Thanks for the clarification!